crypto/punycode.c

```diff
@@ -244,43 +245,34 @@ static int codepoint2utf8(unsigned char *out, unsigned long utf)
  * -1 - bad string passed
  */
 
-int ossl_a2ulabel(const char *in, char *out, size_t *outlen)
+int ossl_a2ulabel(const char *in, char *out, size_t outlen)
```

Comment:
If I remember correctly, the logic I intended here was the outlen is the return value - e.g. if passed buffer is not enough, we could reallocate it and repeat the operation. But I may be wrong.

Reply:
Yeah, that was what was implemented but it's nasty to maintain such semantics using WPACKET.

The alternative would be to keep the returned length but only return the required size if `out == NULL`.  For `out != NULL` return the used size or that the buffer wasn't large enough (with no indication as to how large it needs to be).  The calling would then be:

```c
ossl_a2ulabel(in, NULL, &size);
out = malloc(size);
ossl_a2ulabel(in, out, &size);
```

The two places this function is called don't check the returned size, so I figured it was better to simplify the code and not worry about the `out == NULL` case or return the required size on overflow.

I can change things if it's deemed desirable but, IMO, it would be best if we required the functionality first.

Reply:
Yes, but we do not need this in the current code. If we ever need this, we can add it. This is internal thing.